src/components/button/index.jsx

```diff
@@ -0,0 +1,194 @@
+
+import theme from '../../theme';
+
+const ColoredButton = styled.button`
+    & {
+    	color: ${(props) => (props.colorAlt ? props.colorAlt : theme.white)};
+        min-width: ${(props) => props.minWidth};
+        background-color: ${(props) => props.color};
+        cursor: pointer;
+        font-weight: bold;
+        border: 1px solid ${(props) => props.color};
+
+        &:hover {
+            background-image: ${(props) =>
+                `radial-gradient(circle, transparent 1%, ${darken(0.2, props.color)} 1%)`};
+            background-color: ${(props) => darken(0.2, props.color)};
+        }
+
+        &:active {
+            background-color: ${(props) => darken(0.1, props.color)};
+        }
+
+        &.btn--border {
+            background: ${(props) => (props.colorAlt ? props.colorAlt : 'transparent')};
+            color: ${(props) => props.color};
+            border: 2px solid ${(props) => props.color};
+        }
+
+        &.disabled,
+        &:disabled {
+            background: ${theme.alto};
+            border-color: ${theme.alto};
+            color: ${theme.black};
+            font-weight: normal;
+
+            &:hover,
+            &:active {
+                background: ${theme.alto};
+                border-color: ${theme.alto};
+                color: ${theme.black};
+            }
+
+            &.disabled--clickable:hover {
+                background-image: radial-gradient(
+                    circle,
+                    transparent 1%,
+                    ${darken(0.2, `${theme.alto}`)} 1%
+                );
+                background-color: ${darken(0.2, `${theme.alto}`)};
+                border-color: ${darken(0.2, `${theme.alto}`)};
+            }
+
+            &.disabled--clickable:active {
+                background-color: ${darken(0.1, `${theme.alto}`)};
+                border-color: ${darken(0.2, `${theme.alto}`)};
+            }
+        }
+    }
+
+    &.btn--link {
+        background: transparent;
+        color: ${(props) => props.color};
+        min-width: 0;
+
+        &:hover,
+        &:focus,
+        &:active {
+            background: transparent;
+            box-shadow: none;
+            color: ${(props) => darken(0.2, props.color)};
+            transition: none;
+        }
+    }
+`;
+
+const Content = styled.span`
+    display: flex;
+    align-items: stretch;
+    justify-content: center;
+`;
+
+const Icon = styled.span`
+    display: flex;
+    align-items: center;
+    order: ${(props) => (props.iconOnRight ? '2' : '0')};
+    padding-right: ${(props) => (props.iconOnRight ? '0' : '5px')};
+    padding-left: ${(props) => (props.iconOnRight ? '5px' : '0')};
+`;
+Icon.displayName = 'Icon';
+
+const Text = styled.span`
+    display: flex;
+    align-items: center;
+    padding-left: ${(props) => (props.iconOnRight ? '5px' : '0')};
+	padding-right: ${(props) => (props.iconOnRight ? '0' : '5px')};
+`;
+Text.displayName = 'Text';
+
+export class Button extends Component {
```

Comment:
Probably a next time thing (not a today thing!) but people these days are using less class based components, and more functional components with hooks. I wouldn't worry about this, but you asked for feedback for next time 👍

Reply:
I have a few functional components that use hooks as well in this project. You can't directly test `wrapper.state().stateName` in functional components using hooks yet -- so I've left these two as class components for now, but there are other components such as `save-key` in this PR that are functional components that use hooks as well to show I can make both